test/distributed/checkpoint/test_state_dict.py

```diff
@@ -190,11 +192,39 @@ def init_model_optim():
 
             if compile_model:
                 dist_model = torch.compile(dist_model)
-            dist_optim = torch.optim.Adam(dist_model.parameters(), lr=1e-3)
+            dist_optim = optimizer_class(dist_model.parameters(), lr=1e-3)
             return orig_model, orig_optim, copy_optim, dist_model, dist_optim
 
         self._test_save_load(init_model_optim)
 
+    @with_comms
```

Comment:
Moving this piece around simply we want to have `test_fsdp` and `test_compiled_fsdp` right after the helper function `_test_fsdp`.